modules/ROOT/pages/6.4.1-release-notes.adoc

```diff
@@ -0,0 +1,1296 @@
+
+In {productname} 6.4.1, this new function allows the foreground and background menu icons to function as expected.
+
+=== Pasting links to text would sometimes not generate the correct undo stack on Safari
+//#TINY-9489
+
+In previous versions of {productname}, pasting links to `text` selections in Safari would sometimes not generate the correct undo stack.
+
+This is due to Safari not supporting the ability to track changes to a page within the editor, which can create issues for users when pasting links to `text` selections, as the undo stack would not be able to recognise the link as a separate element, resulting in the undo action when called, to not remove the hyperlink.
+
+To fix this issue, a workaround was implemented in {productname} 6.4.1, that now bypasses the default paste functionality for this undo function.
+
+In {productname} 6.4.1, pasting links to text selections now generates the correct undo stack on Safari.
+
+=== Toolbar split buttons in *Advanced List* plugin to show the correct state when the cursor is in a checklist
+//#TINY-5167
+
+In previous versions of {productname}, when a cursor was placed within a checklist item, the icon for both the checklist and the bullet list would show as `active`. This created UX issues, as having both buttons showing an `active` state was confusing for the user, specifically for navigation purposes when using checklists from the *Advanced List* plugin.
+
+To fix this issue, {productname} added improvements to the list configuration to checks a new set of conditionals which would determine if the `node` selected matched the parent `node` within the checklist. If the match returns as `true`, the (api) will set only the checklist icon to a `active` state.
+
+As a result, the checklist `icon` in {productname} 6.4.1, related to the `advlist` plugin will only appear in a `active` state, when the `node` checklist meets the new conditions implemented.
+
+=== Dragging transparent elements into transparent blocks elements could produce invalid nesting of transparents
+//#TINY-9231
+
+Dragging or dropping block `links` into other `links` was producing invalid nesting of other transparents such as `anchors`. As a result, these transparent elements would produce `anchor` tags within other `anchor` tags, which is not a general practice. As an example `links` and `anchors` that are defined to an element, must not be nested within or contain any other element and vice versa.
+
+Changes were implemented in {productname} 6.4.1 to fix this issue to ensure that any invalid child anchor elements that try to reproduce this action are unwrapped when the editor verifies another transparent has been dropped into another transparent (anchor) element.
+
+In {productname} 6.4.1, dragging content such as `links` into other `links` will now produce *valid* HTML `output` after a drag operation occurs which will be consistent across all browsers.
```

Comment:
```suggestion
In previous versions of {productname}, dragging and dropping inline anchor links into block anchor links was producing invalid nesting of anchors in Chrome. Anchors, like all other transparent elements, must not be nested within each other.

Changes were implemented in {productname} 6.4.1 such that when an inline transparent element such as an anchor is dragged and dropped into a transparent block element of the same type (e.g. also an anchor), the dropped inline transparent element is unwrapped to prevent the nesting behavior. This ensures that the HTML output after the drag operation is valid and consistent across browsers.
```